src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,117 @@
+package duke.ui;
+import duke.task.Task;
+import duke.task.TaskList;
+
+import java.util.Scanner;
+
+/**
+ * A class that deals with all interactions with the user.
+ */
+public class Ui {
+    
+    private static final String MESSAGE_GREETING = "Hello~ I'm Duke!\n" + "What can I do for you?";
+    private static final String MESSAGE_FAREWELL = "Goodbye~";
+    private static final String MESSAGE_DONE = "Nice! I've set this task as done~";
+    private static final String MESSAGE_ADD_TASK = "Got it~ I've added this task:";
+    private static final String MESSAGE_NUMBER_OF_TASKS = "You now have %d tasks in the list~";
+    private static final String MESSAGE_REMOVE_TASK = "Alright~ I've removed this task:";
+    private static final String MESSAGE_SEPERATOR = "____________________________________________________________";
+    private static final String MESSAGE_LIST = "Here are your tasks~";
+    private static final String MESSAGE_FIND = "Here are the matching tasks in your list~";
+    
+    private Scanner sc;
+    
+    
+    public Ui() {
+        sc = new Scanner(System.in);
+    }
+
+    /**
+     * A function to print the welcome message when starting Duke.
+     */
+    public void greet() {
+        System.out.println(MESSAGE_GREETING);
+    }
+
+    /**
+     * A function to print the goodbye message when stopping Duke.
+     */
+    public void farewell() {
+        System.out.println(MESSAGE_SEPERATOR + "\n" + MESSAGE_FAREWELL + "\n" + MESSAGE_SEPERATOR);
+    }
+
+    /**
+     * A function to print the message when marking a task as done.
+     * @param doneTask the Task marked as done.
+     */
+    public void doneText(Task doneTask) {
+        System.out.println(MESSAGE_SEPERATOR + "\n" + MESSAGE_DONE + "\n" + doneTask + 
+                "\n" + MESSAGE_SEPERATOR);
+    }
```

Comment:
I like how you break the line, you think can put the `+` operator at the beginning of the new line instead of the end of the previous line.
More info can be found here under the `Place line to improve readability` section :D 
[https://se-education.org/guides/conventions/java/intermediate.html#comments](url)

Reply:
Yep! Still not used to this mod's coding conventions haha